pkg/ibmvpc/sg_analysis.go

```diff
@@ -111,45 +111,95 @@ func (sga *SGAnalyzer) getRemoteCidr(remote vpc1.SecurityGroupRuleRemoteIntf) (*
 	return target, cidrRes, nil
 }
 
+func (sga *SGAnalyzer) getLocalCidr(local vpc1.SecurityGroupRuleLocalIntf) (*ipblock.IPBlock, string, error) {
+	// TODO: on actual run from SG example, the type of remoteObj is SecurityGroupRuleRemote and not SecurityGroupRuleRemoteCIDR,
+	// even if cidr is defined
+	var localIP *ipblock.IPBlock
+	var cidr string
+	var cidrRes string
+	var err error
+	//TODO: handle other remote types:
+	// SecurityGroupRuleRemoteIP
+	// SecurityGroupRuleRemoteSecurityGroupReference
+
+	// how can infer type of remote from this object?
+	// can also be Address or CRN or ...
+	if localObj, ok := local.(*vpc1.SecurityGroupRuleLocal); ok {
+		switch {
+		case localObj.CIDRBlock != nil:
+			cidr = *localObj.CIDRBlock
+			localIP, err = ipblock.FromCidr(cidr)
+			if err != nil {
+				return nil, "", err
+			}
+			cidrRes = localIP.ToCidrList()[0]
+		case localObj.Address != nil:
+			address := *localObj.Address
+			localIP, err = ipblock.FromIPAddress(address)
+			if err != nil {
+				return nil, "", err
+			}
+			cidrRes = localIP.ToCidrList()[0]
+		default:
+			return nil, "", fmt.Errorf("sg error: getLocalCidr - SecurityGroupRuleLocal is empty: %+v", localObj)
+		}
+
+		if localIP == nil || cidrRes == "" {
+			return localIP, cidrRes, fmt.Errorf("sg error: getLocalCidr returns empty result. localObj: %+v", localObj)
+		}
+	}
+	if localIP == nil || cidrRes == "" {
+		return localIP, cidrRes, fmt.Errorf("sg error: getLocalCidr returns empty result. could not convert localObj to expected type ")
```

Comment:
similar to getRemoteCidr, can avoid code duplication and a separate function with common code?